core/account-id/src/errors.rs

```diff
@@ -1,14 +1,16 @@
 use std::fmt;
 
 /// An error occurred when parsing an invalid Account ID with [`AccountId::validate`](crate::AccountId::validate).
-#[derive(Eq, Hash, Clone, Debug, PartialEq)]
+#[derive(Eq, Clone, Debug, PartialEq)]
 pub struct ParseAccountError(pub(crate) ParseErrorKind, pub(crate) String);
 
 impl ParseAccountError {
+    /// Returns the corresponding [`ParseErrorKind`] for this error.
     pub fn kind(&self) -> &ParseErrorKind {
         &self.0
     }
 
+    /// Returns the corresponding [`AccountId`](crate::AccountId) for this error.
     pub fn get_account_id(self) -> String {
```

Comment:
This method should just not exist, the errors should not store the data which is available to the caller anyway.

Reply:
Bellow: 

```rust
write!(f, "[{}]: {}", self.1, self.0)
```

this is somewhat unusal formatting. I think that's fine probably, as we can always change Display details. Though my gut feeling would be again to try to mimic ParseIntError

Reply:
Yeah, we initially didn't have this, at least not until testing a bunch of IDs in a for-loop was failing, and I needed to inspect which one caused it. But that's a specific use case, and we can drop that since, ideally, the caller should be able to inspect the IDs before parsing.